commonmark-ext-heading-anchor/src/main/java/org/commonmark/ext/heading/anchor/internal/HeadingIdAttributeProvider.java

```diff
@@ -12,12 +12,20 @@
 
     private final IdGenerator idGenerator;
 
-    private HeadingIdAttributeProvider() {
-        idGenerator = IdGenerator.builder().defaultId("heading").build();
+    private HeadingIdAttributeProvider(String defaultId, String prefix, String suffix) {
+        idGenerator = IdGenerator.builder()
+                .defaultId(defaultId)
+                .prefix(prefix)
+                .suffix(suffix)
+                .build();
     }
 
     public static HeadingIdAttributeProvider create() {
```

Comment:
I'm pretty sure i should remove this `create` method. This isn't part of the public API, and will be unused if using the public API